zip-0227.rst

```diff
@@ -1,8 +1,455 @@
+
+3. The issuance validating key, denoted as :math:`\mathsf{ik}`, is the key that is used to validate the issuance transaction. This key is used to validate the issuance of a specific Asset Identifier, and is used by all block chain users (specifically the Asset owners and consensus validators) to associate the Asset in question with the issuer.
+
+The relations between these keys are shown in the following diagram:
+
+.. figure:: zip-0227-key-components-zsa.png
+    :width: 450px
+    :align: center
+    :figclass: align-center
+
+    Diagram of Issuance Key Components for the ZSA Protocol
+
+
+Issuance Authorization Signature Scheme
+```````````````````````````````````````
+
+We define the issuance authorization signature scheme :math:`\mathsf{IssueAuthSig}` similar to :math:`\mathsf{SpendAuthSig}^{\mathsf{Orchard}}`, the Orchard spend authorization signature scheme [#protocol-concretespendauthsig]_. 
+Specifically, we instantiate :math:`\mathsf{IssueAuthSig}` as :math:`\mathsf{RedPallas}` without key re-randomization using generator :math:`\mathcal{P}_{\mathbb{G}} = \mathcal{G}^{\mathsf{Issuance}} := \mathsf{GroupHash}^\mathbb{P}(\texttt{"z.cash:ZSA"}, \texttt{"Issuance"})` where :math:`\mathsf{GroupHash}^\mathbb{P}` is defined as in the Zcash protocol specification [#protocol-concretegrouphashpallasandvesta]_.
+
+Issuance Key Derivation
+```````````````````````
+
+The issuance master key is generated by choosing a bit sequence uniformly at random from :math:`\mathbb{B}^{\mathbb{Y}[32]}`, like the Orchard spending key [#protocol-orchardkeycomponents]_.
+
+Issuance master key derivation for hierarchical deterministic wallets
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''' 
+
+In ``zcashd``, this key is derived in a manner similar to the generation of the Orchard master key in ZIP 32 [#zip-0032-orchard-master]_. We reuse the functions defined there in what follows in this section.
+
+We define the master extended issuance key :math:`m_{\mathsf{Issuance}} := \mathsf{MasterKeyGen}(\texttt{"ZIP32ZSAIssuance"})`.
+
+As in ZIP 32 for Orchard [#zip-0032-orchard-child-key-derivation]_, we only use hardened child key derivation for the issuance master key. 
+We reuse the following function for child key derivation, with a unique personalization value of :math:`[\mathtt{0xa1}]`:
+
+:math:`\mathsf{CDKsk}((\mathsf{sk}_{par},\mathsf{c}_{par}), [\mathtt{0xa1}], i) \to (\mathsf{sk}_i, \mathsf{c}_i)`
+
+We use the notation of ZIP 32 [#zip-0032-orchard-key-path]_ for shielded HD paths, and define the issuance master key path as :math:`m_\mathsf{Issuance} / purpose' / coin\_type' / account'`. We fix the path levels as follows:
+
+- :math:`purpose`: a constant set to :math:`227'` (or :math:`\texttt{0x800000e3}`) following the BIP 43 recommendation.
+- :math:`coin\_type`: Defined as in ZIP 32 [#zip-0032-key-path-levels]_.
```

Comment:
Let's specify the exact value.

Reply:
str4d and Daira had recommended against putting in the explicit `0x80000085` value in the earlier meeting, because someone who forks the repo should know not to hardcode it. They had said it would need rewording in ZIP 32 to specify it is to be set to the base cryptocurrency (aka ZEC for us), so I have left it to be the same as in ZIP 32 for now.